src/main/java/com/minecolonies/apiimp/initializer/ModCraftingTypesInitializer.java

```diff
@@ -0,0 +1,38 @@
+package com.minecolonies.apiimp.initializer;
+
+import com.minecolonies.api.crafting.ModCraftingTypes;
+import com.minecolonies.api.crafting.RecipeCraftingType;
+import com.minecolonies.api.crafting.registry.CraftingType;
+import com.minecolonies.coremod.recipes.BrewingCraftingType;
+import net.minecraft.item.crafting.IRecipeType;
+import net.minecraftforge.event.RegistryEvent;
+import net.minecraftforge.registries.IForgeRegistry;
+
+public final class ModCraftingTypesInitializer
+{
+    private ModCraftingTypesInitializer()
+    {
+        throw new IllegalStateException("Tried to initialize: ModCraftingTypesInitializer but this is a Utility class.");
+    }
+
+    public static void init(final RegistryEvent.Register<CraftingType> event)
+    {
+        final IForgeRegistry<CraftingType> reg = event.getRegistry();
+
+        ModCraftingTypes.SMALL_CRAFTING = new RecipeCraftingType<>(ModCraftingTypes.SMALL_CRAFTING_ID,
+                IRecipeType.CRAFTING, r -> r.canCraftInDimensions(2, 2));
+        reg.register(ModCraftingTypes.SMALL_CRAFTING);
+
+        ModCraftingTypes.LARGE_CRAFTING = new RecipeCraftingType<>(ModCraftingTypes.LARGE_CRAFTING_ID,
+                IRecipeType.CRAFTING, r -> r.canCraftInDimensions(3, 3)
+                    && !r.canCraftInDimensions(2, 2));

```

Comment:
this seems weird, this one can also do 2x2

Reply:
Yes, but that's already covered by the `SMALL_CRAFTING`.  The large crafting module always does both types, but each type has an exclusive subset of recipes with no overlaps, otherwise it would count them more than once.

Reply:
Couldn't we make it then so that the code that checks has either LARGE or SMALL but not together? That seems like it would be cleaner than this

Reply:
We could, but there's a few places that check for only `SMALL_CRAFTING` because they assume that anything that has large crafting does both, and they don't care what crafting size, only that it's vanilla crafting.  If we were to make them mutually exclusive then they'd have to explicitly check for both, which is more wordy.  This just makes everything simpler.

Reply:
Yeah, I know that I had done that in mine too, but now I think it makes more sense to separate that better. There is also a few calls where we check both in some way too, so overall the complexity should stay the same. Technically in some of those places we can even check if IRecipeType.Crafting instead of both (as they got that in common)

Reply:
I disagree.  And there are no places where we check more than one type at once.  It is much much simpler and more sensible the way it is now.